text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,153 @@
+
+### Tier-1 Signals
+
+Similar to a Tier-1 Component, a Tier-1 Signal must have stable support in order to declare OpenTelemetry as stable. Declaring a signal as Tier-1 means that all projects must also see that signal as required for declaring the project stable.
+
+Not all signals will have the same requirements. For instance, Logs might not provide an instrumentation API specification.
+
+List of Tier-1 signals and their capabilities:
+
+* Logs
+  * Data model
+  * SDK specification
+  * Instrumentation SDKs
+  * Bridge API specification
+* Metrics
+  * Data model
+  * API specification
+  * Instrumentation APIs
+  * SDK specification
+  * Instrumentation SDKs
+* Traces
+  * Data model
+  * API specification
+  * Instrumentation APIs
+  * SDK specification
+  * Instrumentation SDKs
+
+The lifecycle of a signal is defined in the [Versioning and stability for OpenTelemetry clients](https://opentelemetry.io/docs/reference/specification/versioning-and-stability). While the stability and versioning of the signals may relate to this OTEP, the previously linked document specifies the details on how to reach the "stable" level mentioned in this OTEP.
+
+The "experimental" level from the signal versioning and stability document relates to the "alpha" and "beta" levels from this OTEP, "stable" and "deprecated" are equivalent to their counterparts on this document, and "removed" is non-existent here.
+
+When a new signal reaches the "stable" level, OpenTelemetry TC and GC will ultimately decide whether to include it as part of OpenTelemetry's priority components, using the process described in [Promoting SIGs and components](#promoting-sigs-and-components).
+
+### Promoting SIGs and components
+
+SIGs can request to be included in the Tier-1 List by contacting a TC or GC member. The TC/GC member will bring the request to the TC/GC via Slack or Zoom call, who will vote on the matter. A simple majority suffices for including the SIG. The SIG needs to have at least equal maturity as the project. For instance, if OpenTelemetry's maturity is set to Beta, the coming SIG has to be at least Beta as well, so that it doesn't downgrade the main project's maturity.
+
+Components can request project maintainers to be included in the Tier-1 list via a GitHub issue created on the main repository of the SIG. The SIG maintainers will then vote on the matter. A simple majority suffices for including the component. The component needs to have at least equal maturity as the SIG so that it doesn't downgrade the SIG's maturity. If SIG maintainers have strong reasons to include a component with a lower status, they can request an exception with the GC/TC. A supermajority vote is required to accept the request until the project reaches the "Stable" level so the project's scope doesn't continuously grow. After we reach the "Stable" level, a simple majority decision will be sufficient.
```

Comment:
>If SIG maintainers have strong reasons to include a component with a lower status, they can request an exception with the GC/TC. A supermajority vote is required to accept the request until the project reaches the "Stable" level so the project's scope doesn't continuously grow. After we reach the "Stable" level, a simple majority decision will be sufficient.

This feels wrong to me.  It seems like it should be the other way around and that it should become significantly more difficult to add a component with a lower stability level once the `Stable` level is reached.  Always requiring a supermajority would also make sense to me.

Is "supermajority" defined somewhere in the governing documents that can be referenced here?  That term could have a wide variety of interpretations.

Reply:
> It seems like it should be the other way around and that it should become significantly more difficult to add a component with a lower stability level once the Stable level is reached

Both would require a vote at the GC/TC level, so it's already hard. I think we had simple majority for both at first, but to avoid feature creep (to use @dyladan's term), we changed so that an approval by 2/3s of the GC/TC members would be required before we get into a stable state.

I'm good either way: making it a simple or super majority for all cases.

@dyladan, what's your opinion on this?

Reply:
About defining supermajority: we don't have this definition anywhere, I'm afraid. I thought this was a well-understood concept, but Wikipedia lists three possible values: 66%, 60%, and 55%. I meant 66% (2/3s) here, I'll add a definition somewhere in this doc.

Reply:
I also thought it was a well understood concept but I guess not. @jpkrohling is right this was originally my idea and it was to prevent too many new things being added that would delay graduation. Maybe we can keep it as a simple majority of both GC and TC (separate votes as they represent separate interests in their decision making). I think everyone involved there is aware of the risk of feature creep delaying graduation.

Reply:
I changed this to be a simple majority.